### ISSUE TITLE
Service Implementation Leads to Contains Unknown Lookup  Error

### ISSUE THREAD

**Issue by nesl9** [abaplint/abaplint-sci-client#242]
I created a service and the structure referenced for the entity and entitySet seem to cause the error below. 

Auto generated service implementation for MPC
```
CLASS zcl_customer_mpc DEFINITION
  PUBLIC
  INHERITING FROM /iwbep/cl_mgw_push_abs_model
  CREATE PUBLIC .

  PUBLIC SECTION.

    TYPES:
       ts_customer TYPE zs_customer. .
    TYPES:
  tt_customer TYPE STANDARD TABLE OF ts_customer. .
```

Leads to error:
`Type "TS_CUSTOMER" contains unknown: ZS_CUSTOMER not found, lookup, ZCL_CUSTOMER_MPC.CLAS.ABAP`

ZS_CUSTOMER is globally defined in the ABAP Dictionary.

A similar issue occurs with VBAK when referencing it in a data type, in this case "lookupDomain" as opposed to "lookup" 
`DATA: t_vbak      TYPE HASHED TABLE OF vbak WITH UNIQUE KEY vbeln.`

Leads to error:
`Variable "LT_VBAK" contains unknown: ZUONR, lookupDomain (unknown_types)`

**Comment by larshp** [comment]
the sci client has multiple options, what is the dependency depth set to?

If its just 1, then the VBAK table will be sent to the server, but not the data elements and domains for each table line

do consider how much or little you want to syntax check it against the standard code

report ZABAPLINT_DEBUG can be used to download a zip of the actual content/objects being sent to the server

![image](https://user-images.githubusercontent.com/5888506/109051440-d4cbb480-76da-11eb-8d8a-fee443057c13.png)

**Comment by larshp** [comment]
when running with SCI, the dependecies from abaplint.json are ignored, as they are instead fetched from the system

**Comment by nesl9** [comment]
I see. Currently my configuration is identical to yours with the exception of the URL that points to my private server running abaplint: 2.69.3. I'll play around with these settings to understand them better. 

I created repo: https://github.com/10xcoding/sap_service_example.git
![image](https://user-images.githubusercontent.com/1100456/109052397-2fe0b580-76a1-11eb-9fd5-8b34c6249c5b.png)

Thank you very much for all the feedback!

**Comment by nesl9** [comment]
Update --
By selecting the "Continue Into SAP Objects" checkbox that fixed the issue with VBAK-ZUONR.
![image](https://user-images.githubusercontent.com/1100456/109104751-30556c80-76f2-11eb-9e3b-aae6b9c0de3a.png)

As for the issue with the unknown structure, I updated the repo (https://github.com/10xcoding/sap_service_example.git) to include four test cases:
1. Entity from SAP standard structure E1KNA1C
2. Entity from SAP standard table KNA1
3. Entity from custom structure ZCUSTOMER_S
4. Entity from custom table ZCUSTOMER_T
![image](https://user-images.githubusercontent.com/1100456/109104739-292e5e80-76f2-11eb-8868-ef4c3b444b4b.png)

It seems the error only shows up for 3 and 4. Changing the depth led to other errors. Any suggestions on what to look into? 
![image](https://user-images.githubusercontent.com/1100456/109104903-80343380-76f2-11eb-9d6f-dba8633db202.png)

```
  PUBLIC SECTION.

    TYPES:
       ts_customerzstructure TYPE zcustomer_s. .
    TYPES:
  tt_customerzstructure TYPE STANDARD TABLE OF ts_customerzstructure. .
    TYPES:
      BEGIN OF ts_text_element,
        artifact_name        TYPE c LENGTH 40,       " technical name
        artifact_type        TYPE c LENGTH 4,
        parent_artifact_name TYPE c LENGTH 40, " technical name
        parent_artifact_type TYPE c LENGTH 4,
        text_symbol          TYPE textpoolky,
      END OF ts_text_element. .
    TYPES:
           tt_text_elements TYPE STANDARD TABLE OF ts_text_element WITH KEY text_symbol. .
    TYPES:
       ts_customerztable TYPE zcustomer_t. .
    TYPES:
  tt_customerztable TYPE STANDARD TABLE OF ts_customerztable. .
    TYPES:
       ts_customersaptable TYPE kna1. .
    TYPES:
  tt_customersaptable TYPE STANDARD TABLE OF ts_customersaptable. .
    TYPES:
       ts_customersapstructure TYPE e1kna1c. .
    TYPES:
  tt_customersapstructure TYPE STANDARD TABLE OF ts_customersapstructure. .

    CONSTANTS gc_customersapstructure TYPE /iwbep/if_mgw_med_odata_types=>ty_e_med_entity_name VALUE 'CustomerSapStructure' ##NO_TEXT.
    CONSTANTS gc_customersaptable TYPE /iwbep/if_mgw_med_odata_types=>ty_e_med_entity_name VALUE 'CustomerSapTable' ##NO_TEXT.
    CONSTANTS gc_customerzstructure TYPE /iwbep/if_mgw_med_odata_types=>ty_e_med_entity_name VALUE 'CustomerZStructure' ##NO_TEXT.
    CONSTANTS gc_customerztable TYPE /iwbep/if_mgw_med_odata_types=>ty_e_med_entity_name VALUE 'CustomerZTable' ##NO_TEXT.
```

**Comment by nesl9** [comment]
Just to close the loop on this and clarify the problem. I got inconsistent results in GitHub with the abaplint app vs in ECC with the abaplint-sci-client. 

With the GitHub App: abaplint.json used the default deps but didn't produce an error.

With SCI: after some debugging of the client and based on your response above, SCI produces the error (Type "TS_CUSTOMER" contains unknown: ZS_CUSTOMER not found, lookup, ZCL_CUSTOMER_MPC.CLAS.ABAP) when a Z data type is referenced in a gateway service entity. The error goes away if I modify ZCL_ABAPLINT_DEPS_FIND->RESOLVE to include my Z data type as a dependency. 

```
" Test dependencies
IF ls_wbcrossgt-otype EQ 'TY' AND ls_wbcrossgt-name(1) EQ 'ZCUSTOMER'.
      CLEAR ls_tadir.
      ls_tadir-ref_obj_type = 'TABL'.
      ls_tadir-ref_obj_name = ls_wbcrossgt-name.
      INSERT ls_tadir INTO TABLE rt_tadir.
ENDIF.
```
![image](https://user-images.githubusercontent.com/1100456/109360253-7a526580-784c-11eb-801a-4c50419cac80.png)

This now makes sense to me but I would've expected the GitHub App to also produce the error unless I added custom deps.
